### PR TITLE
Allow to parse JSON objects into Strings in JSON input formats

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a new setting `input_format_json_read_objects_as_strings` that allows to parse nested JSON objects into Strings in all JSON input formats. This setting is disable by default.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by tavplubix** [comment]
What's the difference from `JSONAsString`?

**Comment by Avogar** [comment]
`JSONAsString` allows to read the whole JSON object into one single column with type `String`.
For example:
`{"x" : {"a" : 1}, "y" : 1}` will be read into one column with value `'{"x" : {"a" : 1}, "y" : 1}'`.
This new setting will allow to read nested JSON objects into String column.
For example:
`{"x" : {"a" : 1}, "y" : 1}` can be parsed into two columns `x String, y UInt64` with values `'{"a" : 1}'` and `1` using `JSONEachRow` format.
I think it will be useful, because `JSON` type is still experimantal, so nested JSON objects can be parsed into a string and then processed using `JSONExtract*` functions. Sometimes I see that people in our Cloud have data with nested objects, but cannot process it without experimantal `JSON` type

File: src/Core/Settings.h [hunk 1]
```diff
@@ -546,7 +546,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     \
     M(Bool, collect_hash_table_stats_during_aggregation, true, "Enable collecting hash table statistics to optimize memory allocation", 0) \
     M(UInt64, max_entries_for_hash_table_stats, 10'000, "How many entries hash table statistics collected during aggregation is allowed to have", 0) \
-    M(UInt64, max_size_to_preallocate_for_aggregation, 10'000'000, "For how many elements it is allowed to preallocate space in all hash tables in total before aggregation", 0) \
+    M(UInt64, max_size_to_preallocate_for_aggregation, 100'000'000, "For how many elements it is allowed to preallocate space in all hash tables in total before aggregation", 0) \
```
Comment (Avogar) [inline src/Core/Settings.h:549]: This was added by mistake, will fix it

Comment (Avogar) [reply src/Core/Settings.h:549]:
```suggestion
    M(UInt64, max_size_to_preallocate_for_aggregation, 10'000'000, "For how many elements it is allowed to preallocate space in all hash tables in total before aggregation", 0) \
```

**Comment by tavplubix** [comment]
Thanks for the explanation, it was not obvious from the example in documentation, because there's only one column. Consider updating the example.

**Comment by Avogar** [comment]
Agree, example is not good, I will update it

**Review by tavplubix — APPROVED** [review]

**Comment by Avogar** [comment]
`AST fuzzer (tsan)` - https://github.com/ClickHouse/ClickHouse/issues/43687
`AST fuzzer (ubsan)` - https://github.com/ClickHouse/ClickHouse/issues/44084

**Pull request merged**